rosetta-ide/src/main/java/com/regnosys/rosetta/ide/server/RosettaLanguageServerImpl.java

```diff
@@ -185,4 +217,44 @@ public CompletableFuture<FormattingOptions> getDefaultFormattingOptions() {
 			return CompletableFuture.failedFuture(e);
 		}
 	}
+	
+	@Override
+	public CompletableFuture<CodeAction> resolveCodeAction(CodeAction unresolved) {
+		return getRequestManager().runRead((cancelIndicator) -> resolveCodeAction(unresolved, cancelIndicator));
+	}
+
+	protected CodeAction resolveCodeAction(CodeAction codeAction, CancelIndicator cancelIndicator) {
+		CodeActionParams codeActionParams = codeActionUtils.getCodeActionParams(codeAction);
+
+		if (codeActionParams.getTextDocument() == null) {
+			return null;
+		}
+
+		URI uri = getURI(codeActionParams.getTextDocument());
+
+		IQuickFixProvider quickfixes = getService(uri, IQuickFixProvider.class);
+
+		return getWorkspaceManager().doRead(uri, (doc, resource) -> {
+			ICodeActionService2.Options baseOptions = new ICodeActionService2.Options();
```

Comment:
I think all of this logic should be externalized into its own `IResolveCodeActionService`, similar to other methods in this class.